src/platform/silabs/multi-ota/efr32/OTAHooks.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <platform/silabs/multi-ota/MultiOTAImageProcessorImpl.h>
+#include <include/platform/CHIPDeviceLayer.h>
+
+#include <app/clusters/ota-requestor/OTARequestorInterface.h>
+
+#include <platform/silabs/multi-ota/efr32/OTAFirmwareProcessor.h>
+
+// TODO: Find home for CONFIG_CHIP_SILABS_OTA_FACTORY_DATA_PROCESSOR
+/** 
+ * @def CONFIG_CHIP_SILABS_OTA_FACTORY_DATA_PROCESSOR
+ *
+ * Enables default OTA TLV factory data processor.
+ * Disabled by default.
+ */
+#ifndef CONFIG_CHIP_SILABS_OTA_FACTORY_DATA_PROCESSOR
+#define CONFIG_CHIP_SILABS_OTA_FACTORY_DATA_PROCESSOR 0
+#endif // CONFIG_CHIP_SILABS_OTA_FACTORY_DATA_PROCESSOR
+
+
+#if CONFIG_CHIP_SILABS_OTA_FACTORY_DATA_PROCESSOR
+#include <platform/silabs/multi-ota/efr32/FactoryDataProcessor.h>
+#endif // CONFIG_CHIP_SILABS_OTA_FACTORY_DATA_PROCESSOR
+
+
+extern "C" void ResetMCU(void);
```

Comment:
Why this extern "C" functions? If you need an abstraction to reset platforms it should be a cpp function in Silabs Platform abstraction

Reply:
They don't need to be extern, I removed the extern tag